samza-core/src/main/java/org/apache/samza/clustermanager/FaultDomainManager.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.clustermanager;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * This interface gets fault domain information of different nodes from the cluster manager (Yarn/Kubernetes/etc.).
+ *  It also provides other functionality like exposing all the available fault domains, checking if two hosts belong to
+ *  the same fault domain, and getting the valid fault domains that a standby container can be placed on.
+ */
+public interface FaultDomainManager {
```

Comment:
The interface exposes too many functionalities that are either not necessary upstream or conflating.
 
1. `computeNodeToFaultDomainMap`
2. `getAllowedFaultDomainsForSchedulingContainer`
3. `getNodeToFaultDomainMap`

The notion of allowed fault domain for the container is not determined by the cluster manager's fault domain manager rather `StandbyContainerManager`. Hence [2] shouldn't be exposed as a functionality.
[3] seems redundant given you have `getFaultDomainOfNode`
[1] is completely internal to the manager and shouldn't be exposed.

Reply:
This makes sense. Made the changes. Thanks for the suggestions :)